src/go/k8s/apis/redpanda/v1alpha1/cluster_types.go

```diff
@@ -1130,3 +1130,10 @@ func defaultTLSConfig() *TLSConfig {
 		NodeSecretRef:     nil,
 	}
 }
+
+func (s Cluster) SASLAuthorizationEnabled() bool {
+	return s.Spec.KafkaEnableAuthorization ||
+		s.Spec.EnableSASL ||
+		s.InternalListener().AuthenticationMethod == "sasl" ||
```

Comment:
Only internal listener?

Reply:
The SASLAutorizationEnabled function is used only internally to configure:
* pandaproxy talking to kafka listener
* schema registry talking to kafka listener
* console talking to kafka listener

In the code we have assumption to not talk over external listeners.

Reply:
let's encode that assumption in the method name then? 🙏 `IsSASLOnInternalEnabled` for example